src/server/db_slice.cc

```diff
@@ -601,6 +613,7 @@ void DbSlice::PreUpdate(DbIndex db_ind, PrimeIterator it) {
   }
 
   it.SetVersion(NextVersion());
+  it->first.SetSticky(false);
```

Comment:
>  in case of STICK, I do think it makes sense to destruct this property upon writes 

As far as I could tell, all writes go through `PreUpdate`. I checked a handful of commands & grepped ` it.SetVersion`. Hope there are no exceptions.

Reply:
if I understand correctly, you actually destruct this property, when calling `SetSticky(false)`.
i.e. "set key bar", "stick key", and then "set key bar" will remove stickiness. I actually meant the opposite:
to preserve the stickiness property for all operations.